### PR TITLE
Sprint 1 PR Review Alt

### PR DESCRIPTION
**Pull request by RafterGit**
Adding a template for Pull Requests

###This pull request changes...

Updates in response to the Sprint 1 PR Review:

1.) The CI/CD deployment pipeline has been expanded to now utilize the `deploy-cloudgov-docker.sh` script. With the correct parameters this script will deploy frontend and backend docker images via a rolling up or a standard redeployment.

- This script also facilitates the ability to construct a new deployment environment in cloud.gov if the user so chooses. 

- This script will also leverage the dynamic naming of the docker images to be pushed via the manifest.yml. With this method, we can tag versions for release and circumvent needing to hard code this value in our manifest.yml

2.) Updates to the library vulnerabilities based around snyk have also been added t the CI/CD pipeline via the circleci config.yml. Future updates, we will expand this to include docker image scanning as well. Currently, since there is no way to disable snyk failing if it encounters vulnerabilities with no reported fix, we've disabled the snyk  auto fail feature on the UI test.

https://support.snyk.io/hc/en-us/articles/360001702758-Why-is-an-ignored-vulnerability-still-showing-in-the-CLI-when-running-Snyk-Test-

Screen shot of the  vulnerability with no known fix as defined in the  `.snyk` policy file, still showing up in circleci scans:
Screen Shot 2020-08-27 at 8 55 00 PM

3.) A markdown file has been added to tdrs-backend/docs outlining the current issues with finding a useful OIDC library to Login.gov authentication in lieu of our own.

4.) Added scripts to check if needed commands have been installed and installs them if they're missing. CircleCI should be cacheing some of these installations but it's not at the moment. This will be addressed in a future PR

5.)  CircleCI deployment to cloud.gov :

https://app.circleci.com/pipelines/github/raft-tech/TANF-app/376/workflows/2d8827bb-1a84-4134-87d0-db96925ebdf7/jobs/604

https://tdp-frontend.app.cloud.gov/
###[AC] This pull request is ready to merge when...

- [ ] Tests have been updated (and all tests are passing)
- [ ] Vulnerability Scan Test (Zap) has no high/critcial CVEs
- [ ] Linting Tests (ESLint for UI and Flake8 for Python)
- [ ] Unit Tests ( including a code coverage test report)
- [ ] Accessibility Tests (Pa11y UI Only)
- [ ] This code has been reviewed by someone other than the original author
- [ ] The experience passes a basic manual accessibility audit (keyboard nav, screenreader, text scaling) OR an exemption is documented
- [ ] The change has been documented
- [ ] Associated OpenAPI documentation has been updated
- [ ] Changelog is updated as appropriate

###[DoD] This feature is done when...

- [ ] Design has approved the experience
- [ ] Product has approved the experience

### CODE REVIEW

File: tdrs-frontend/README.md [hunk 1]
```diff
@@ -184,39 +189,31 @@ This section has moved here: https://facebook.github.io/create-react-app/docs/tr
...
 cf target -o <ORG> -s <SPACE>
 ```
 
-4.) Push the image to Cloud.gov ( please ensure you're in the same directory as the manifest.yml):
-
-`cd tdrs-backend; cf push tdp-frontend --docker-image goraftdocker/tdp-frontend:devtest`
+3.) Push the image to Cloud.gov (  you will need to be in the same directory as`tdrs-frontend/manifest.yml`):
 
-5.) You will then have to set all required environment variables via the cloud.gov GUI or CF CLI
+( **The `--var` parameter ingest a value into the ((docker-backend)) environment variable in the manifest.yml**)
```
Comment (carltonsmith) [inline tdrs-frontend/README.md:213]: just a grammar note: ingest should be ingests

File: tdrs-frontend/README.md [hunk 2]
```diff
@@ -184,39 +189,31 @@ This section has moved here: https://facebook.github.io/create-react-app/docs/tr
...
 
- `cf set-env tdp-backend JWT_KEY "$(cat test.txt)"`
- **For the list of required envrionment variables please defer to `tdrs-backend/tdpservice/settings/env_vars/.env.local`
+`cf push tdp-frontend --no-route -f manifest.yml --var docker-backend=goraftdockertdp-frontend:devtest`
 
+4.) To  may be required to restage the application after deployment:
```
Comment (carltonsmith) [inline tdrs-frontend/README.md:217]: "To" looks like it should be "It"

**Comment by lauraGgit** [comment]
Also looks like the most recent push on this branch broke the frontend tests?

**Comment by RafterGit** [comment]
> Also looks like the most recent push on this branch broke the frontend tests?

Looks like an intermittent issue with the zap-scan script on yarn install, I just re-ran the job without any chanes and it completed without issue: 

https://app.circleci.com/pipelines/github/raft-tech/TANF-app/380/workflows/d2c7f1d8-60c7-4ac8-8d22-2a3104891a23/jobs/620/parallel-runs/0/steps/0-110

**Pull request merged**